### ISSUE TITLE
Huge FPS lag on client - down to single digit

### ISSUE THREAD

**Issue by Hiranus** [ShetiPhian/Localizations-and-Issues#346]
**Describe the bug**
Huge fps lags when stuff from this mod and Integrated Dynamics are close together.
Bug on ID side: https://github.com/CyclopsMC/IntegratedDynamics/issues/1155

Build this and watch your fps die.
![obraz](https://user-images.githubusercontent.com/6943031/163567722-0435d4aa-6b1f-4bfe-afa5-770bc2ebdc71.png)

**Additional context**
Minecraft Version: 1.16.5
Forge Version: 36.2.31
ShetiPhian Core Version: 3.8.13
My Mod(s) Name & Version: Ender Chests 1.7.12
Are other mods present?
If yes:
- Does everything work without my mods? If I replace your chests with other stuff I see no fps issues
- Have you tested with only my mods? No, as its issue between mods. Spamming your chests alone (without ID stuff in proximity) doesn't cause fps issues

Without Optifine: https://spark.lucko.me/YCCRJdhnk2
With Optifine: https://spark.lucko.me/vQRKMJAo2p

There are no errors in client log when building this contraption.

**Comment by Hiranus** [comment]
https://gfycat.com/saltyboringborderterrier

**Comment by ShetiPhian** [comment]
I see the other dev is looking into it and believes its recent change they've made.
Just in case it turns out to be something else here's some questions (no need to test or answer until the other fix is tested)

Are the pipes trying to move items in/out of the chests? as they don't appear to be connected.
I ask because when the contents change all chests that share an inventory update the neighbors so things like comparators can work. That layout may be causing the code between the mods to go haywire.
If all the chests are different colors does it still happen?

**Comment by Hiranus** [comment]
When looking at the call stack I also assumed its on ID end, but you were in call stack, so I assumed it would be good idea to notify you too.
It doesn't happen with vanilla chests, furnaces, or even stuff from other mods, that's why I created report here. To me it looks like ID creator made some performance heavy code which causes spike lag, but only once (when something next to cable is changing). Your chests run that ID code every tick, creating super effective fps killer. But there is one thing that is interesting - EnderTanks tank doesn't create that fps killer combo.
About your questions:
No, those pipes are not connected, they are just sitting there doing nothing. In fact server side doesn't care at all what I do, it runs perfectly smoothly. Its client side specific issue. 
I tried it with with chests of different colors and effect is exactly the same.

**Comment by ShetiPhian** [comment]
I didn't notice the two links at the bottom, started reading the first post followed the ID link, came back her looked at your gfycat and responded. -_-

Its a rendering lag issue, so the server is unaware of it. It also indicates what I posted above is incorrect as that is a logic issue and would hit the server, not the client.

Its happening in the BlockEntitiy renderer, which both explains the frequency and why EnderTanks doesn't have the issue.
BlockEntities render every tick and EnderTanks doesn't have any. EnderChests only use them for the lid so it can animate, the body is a block model.

[I've retyped this next part so many times, kept changing it as I thought things through more]

The ID hooks are in the BlockModel, and appear to be a bit process heavy, but that isn't much of an issue because BlockModels are only updated when the chunk updates, and is rerendered every tick from the cache.

BlockEntities on the other hand are processed and rendered every tick while the block is in render range.

My lids are a BlockModel rendered via BlockEntity, and due to that how fast a BlockModel is processed matters much more.

Ideally if I could, I would replicate the chunk update and have the BlockEntity render a cached version except when the block changes, but rendering is by far my weakest, I get lost trying to follow what Minecraft is doing.
